src/Analysis/AtlasDataDictionary/Components/ManageColumns/ManageColumns.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Button,
+  Popover,
+  Switch,
+  Notification,
+  Grid,
+  Space,
+} from '@mantine/core';
+import ManageColumnsIcon from '../Icons/ManageColumnsIcon';
+import RestoreIcon from '../Icons/RestoreIcon';
+import HolderIcon from '../Icons/HolderIcon';
+import { IColumnManagementData } from '../../Interfaces/Interfaces';
+import './ManageColumns.css';
+
+interface IManageColumns {
+  handleTableChange: Function;
+  columnManagementData: IColumnManagementData;
+}
+const ManageColumns = ({
+  handleTableChange,
+  columnManagementData,
+}: IManageColumns) => {
+  const [opened, setOpened] = useState(false);
+  const [showNotification, setShowNotification] = useState(false);
+  const notificationDisplayTimeInMilliseconds = 3000;
+  const iconSpanSize = 1;
+  const columnManagementResetButtonSpanSize = 11;
+  const columnLabelSpanSize = 8;
+  const switchSpanSize = 3;
+
+  useEffect(() => {
+    if (showNotification) {
+      setTimeout(() => {
+        setShowNotification(false);
+      }, notificationDisplayTimeInMilliseconds);
+    }
+  }, [showNotification]);
+
+  useEffect(() => {
+    localStorage.setItem(
+      'atlasDataDictionaryColumnManagement',
+      JSON.stringify(columnManagementData),
+    );
+  }, [columnManagementData]);
+
+  const columnManagementReset = () => {
+    handleTableChange(
+      'columnManagementReset',
+      'columnManagementReset',
+    );
+    setShowNotification(true);
+  };
+
+  const columnControls = [
```

Comment:
this list seems to repeat some of the same attributes of the items in the list in ColumnHeaders. Is there a way to have the list of columns for this table defined once and then perhaps refer to it here?

Reply:
This is a very good point. 

The columnControls have been moved to the utils folder and renamed ```ColumnItems```
This is now used by both the ManageColumns component and the ColumnHeaders component. The ```label``` key value pair is now computed in ManageColumns by stripping out the HTML elements used in the JSX for the ColumnHeaders components, which need line breaks in specific places to match the design. The ```ID``` key value pair is now ```headerKey``` for both components.